silkrpc/commands/eth_api.cpp

```diff
@@ -447,17 +447,34 @@ asio::awaitable<void> EthereumRpcApi::handle_eth_get_uncle_count_by_block_number
 
 // https://eth.wiki/json-rpc/API#eth_gettransactionbyhash
 asio::awaitable<void> EthereumRpcApi::handle_eth_get_transaction_by_hash(const nlohmann::json& request, nlohmann::json& reply) {
+    auto params = request["params"];
+    if (params.size() != 1) {
+        auto error_msg = "invalid eth_getTransactionByHash params: " + params.dump();
+        SILKRPC_ERROR << error_msg << "\n";
+        reply = make_json_error(request["id"], 100, error_msg);
+        co_return;
+    }
+    auto transaction_hash = params[0].get<evmc::bytes32>();
+    SILKRPC_DEBUG << "transaction_hash: " << transaction_hash << "\n";
+
     auto tx = co_await database_->begin();
 
     try {
         ethdb::TransactionDatabase tx_database{*tx};
-
-        reply = make_json_content(request["id"], to_quantity(0));
+        const auto optional_transaction = co_await core::rawdb::read_transaction_by_hash(tx_database, transaction_hash);
+        if (!optional_transaction) {
+            // TODO
+            // Maybe no finalized transaction, try to retrieve it from the pool
+            SILKRPC_DEBUG << "Retrieving not finalized transactions from pool not implemented yet\n";
+            reply = make_json_content(request["id"], nullptr);
+        } else {
+            reply = make_json_content(request["id"], *optional_transaction);
+        }
     } catch (const std::exception& e) {
```

Comment:
First of all exception `std::invalid_argument` should be catched to handle possible failures coming from internal functions, e.g. see [line 104 in chain.cpp](https://github.com/torquem-ch/silkrpc/blob/master/silkrpc/core/rawdb/chain.cpp#L104).

This mechanism (I mean throwing `std::invalid_argument` instead of using `std::optional` for data not found) is far from ideal and will be subject to change in the future, but it allows for clean implementation of higher-level methods (that otherwise would suffer from repeated resource-closing code blocks in absence of RAII support in coroutines).

Please check carefully the return value spec in this case, probably empty value (not error)

Reply:
The `std::optional` management is required as the `core::rawdb::read_transaction_by_hash` should return a null `Transaction` as it could be in not yer finalized block.  For that reason the call does't throw a `std::invalid_argument` exception if a `Transaction` is not found for the given hash.
I could add a catch for `std::invalid_argument` for any future modification (and I did).